File: ride-sharing-system/java-without-multithreading/entities/Vehicle.java
package entities;
import enums.VehicleType;

public class Vehicle {
    private String ownerId;
    private String registrationNo;
    private VehicleType type;

    public Vehicle(String ownerId, String registrationNo, VehicleType type) {
        this.ownerId = ownerId;
        this.registrationNo = registrationNo;
        this.type = type;
    }

    public String getOwnerId() { return ownerId; }
    public String getRegistrationNo() { return registrationNo; }
    public VehicleType getType() { return type; }
}

File: ride-sharing-system/go-without-multithreading/entities/vehicle.go
package entities
import "app/ride-sharing-system/go-without-multithreading/enums"

// Vehicle securely links an owner and physical vehicle mapping structures mathematically.
// Binds strict typing avoiding string manipulation errors natively!
type Vehicle struct {
	OwnerID        string
	RegistrationNo string
	Type           enums.VehicleType
}
